src/common/test/myProfileMocking.ts

```diff
@@ -155,3 +207,189 @@ export const getVerifiedData = (
   },
   ...overrides,
 });
+
+function createNewProfileNode(dataType: EditDataType): MultiItemProfileNode {
+  return {
+    ...formConstants.EMPTY_VALUES[dataType],
+  };
+}
+
+function findNode(
+  nodes: MultiItemProfileNode[],
+  id: string,
+  returnIndex = false
+): number | MultiItemProfileNode | undefined {
+  const func = returnIndex ? 'findIndex' : 'find';
+  return nodes[func](node => node.id === id);
+}
+
+function getEdges(
+  profileData: ProfileData,
+  dataType: EditDataType
+): EdgeList | null {
+  const targetPath = `${dataType}.edges`;
+  return _.get(profileData, targetPath, null);
+}
+
+function getEdgeTypeName(
+  dataType: EditDataType
+): 'EmailNodeEdge' | 'AddressNodeEdge' | 'PhoneNodeEdge' {
+  return dataType === 'emails'
+    ? 'EmailNodeEdge'
+    : dataType === 'phones'
+    ? 'PhoneNodeEdge'
+    : 'AddressNodeEdge';
+}
+
+function getPrimaryPropName(
+  dataType: EditDataType
+): 'primaryAddress' | 'primaryPhone' | 'primaryEmail' {
+  return dataType === 'emails'
+    ? 'primaryEmail'
+    : dataType === 'phones'
+    ? 'primaryPhone'
+    : 'primaryAddress';
+}
+
+function createEdge(
+  dataType: EditDataType,
+  nodeData: Partial<MultiItemProfileNode>
+): Mutable<InsertableEdge> {
+  const typename = getEdgeTypeName(dataType);
+  const node = {
+    ...createNewProfileNode(dataType),
+    ...nodeData,
+  };
+  return {
+    __typename: typename,
+    node,
+  } as InsertableEdge;
+}
+
+function setPrimaryNode(
+  profile: Mutable<ProfileData>,
+  dataType: EditDataType,
+  newPrimary: Mutable<MultiItemProfileNode> | null
+): void {
+  if (newPrimary) {
+    newPrimary.primary = true;
+  }
+  const propName = getPrimaryPropName(dataType);
+  Object.assign(profile, { [propName]: newPrimary });
+}
+
+export function cloneAndManipulateProfile(
```

Comment:
I find the naming of this function little problematic. It does clone the profile, but does not really give the result without calling the additional functions that it returns. It feels more like a class type of object than a function that does a particular thing. Maybe renaming it to _clonedProfileWithManipulationFunctions_ etc would be more describing?

Reply:
"clonedProfileWithManipulationFunctions" sounds like a return type?

Renamed to cloneProfileAndProvideManipulationFunctions (which is a long monster :D ) IMHO the actual return type describes the function already quite well. But that is only seen via hover in the IDE.